### PR TITLE
More linux fixes

### PR DESCRIPTION
**Pull request by mmuman**
* drop -fno-implicit-templates for GNU/Linux
* add test for libGLU (you'll need apt:libglu1-mesa-dev)
* fix build in triangulate.hpp (missing include for memset())

### CODE REVIEW

File: toonz/sources/include/tcg/hpp/triangulate.hpp [hunk 1]
```diff
@@ -15,6 +15,7 @@
 #include <GLUT/glut.h>
 #elif LINUX
 #include <GL/glut.h>
+#include <string.h>
```
Comment (andreygursky) [inline toonz/sources/include/tcg/hpp/triangulate.hpp]: hpp means it is a C++ header, thus cstring should be used

Comment (mmuman) [inline toonz/sources/include/tcg/hpp/triangulate.hpp]: right, I'm used to Haiku conventions I suppose... except it complains with <string>, including when forcing the default namespace with ::memset :^)

Comment (andreygursky) [inline toonz/sources/include/tcg/hpp/triangulate.hpp]: > except it complains with string

I mentioned **c**string.

Comment (mmuman) [inline toonz/sources/include/tcg/hpp/triangulate.hpp]: oh right, my bad, I still don't sleep enough :D

Comment (mmuman) [inline toonz/sources/include/tcg/hpp/triangulate.hpp]: I'll push a rebased branch

File: toonz/sources/include/tsound_t.h [hunk 2]
```diff
@@ -431,7 +431,7 @@ class TSoundTransform
 };
 
 //==============================================================================
-#ifdef MACOSX
+#if defined(MACOSX) || defined(LINUX)
```
Comment (andreygursky) [inline toonz/sources/include/tsound_t.h]: But here you haven't mentioned I fixed this in my PR.

Comment (mmuman) [inline toonz/sources/include/tsound_t.h]: But the commit log was too vague, and the rest of the commit wasn't a fix at all.

**Comment by mmuman** [comment]
So here I have everything building and the main exe running.

**Comment by mmuman** [comment]
@ideasman42  So, I think this first batch of fixes to be good enough, would you mind pulling those so we can all rebase against your branch and continue working? I'll just go get some sleep, should already be sleeping for hours :D

**Comment by ideasman42** [comment]
Thanks for all the improvements, merged - though amended one minor indentation issue (glutInit), now Im getting `This branch has conflicts that must be resolved`  message from github.

So closing, since in practice this is merged.

**Comment by mmuman** [comment]
Thanks

**Comment by mmuman** [comment]
Yeah as you did the merge manually and changed a commit, github is just lost as to whether it's merged or not.

**Pull request closed without merging**